core/fifo_cache.go

```diff
@@ -7,52 +7,69 @@ import (
 	"sync"
 )
 
-const minCacheSize = 16
+var (
+	_ FIFOCache[int, int] = (*BufferFIFOCache[int, int])(nil)
+	_ FIFOCache[int, int] = (*NoOpFIFOCache[int, int])(nil)
+)
 
 // FIFOCache evicts the oldest element added to it after [limit] items are
 // added.
-type FIFOCache[K comparable, V any] struct {
-	limit int
-
-	l sync.RWMutex
-
-	buffer *BoundedBuffer[K]
-	m      map[K]V
+type FIFOCache[K comparable, V any] interface {
+	Put(K, V)
+	Get(K) (V, bool)
 }
 
 // NewFIFOCache creates a new First-In-First-Out cache of size [limit].
 //
-// If [limit] is less than 16, [limit] will be overwritten with 16.
-func NewFIFOCache[K comparable, V any](limit int) *FIFOCache[K, V] {
-	if limit < minCacheSize {
-		limit = minCacheSize
+// If a [limit] of 0 is passed as an argument, a no-op cache is returned that
```

Comment:
Some tests run with `AcceptedCacheSize=0`. We can either:
* Pipe a `nil` check on this cache through the code (gross)
* Ensure all configs run with a cache > 0
* Enforce a min cache size (to overwrite 0 limit)
* Do this

Reply:
This looks nice.